### DISCUSSION TITLE
Localize the Wagtail Image library

### DISCUSSION THREAD

**Discussion by enzedonline** [wagtail/wagtail-localize#378] [Ideas]
Not sure if this is on the plan - 

Would be great to be able to have translated titles in the image gallery so the alt={{ image.title }} shows the title in the language of the page. At the moment I'm using an image chooser struct block for this, but it means the image title has to be entered and translated each time it's used.

**Comment by kaedroho** [comment]
Hi @enzedonline

Yep, this currently works at the model level: you can add ``TranslatableMixin`` to your image model, and Wagtail Localize will automatically translate any foreign keys that point to images to their translated versions.

What's missing here is on the view level: We need to add the ability to filter by Locale in the Wagtail UI, and use Wagtail Localize to actually do the translation

**Reply by enzedonline** [reply]
Hi @kaedroho 

Great, thanks, I'll look into that. Is this one image per language or just one image with translated titles?

I was thinking to be able to click into the edit page of an image in the gallery, select language and then translate the title - image.title then just pulls in the relevant translated title without need for multiple images.

Filtering in the UI would be great - I'm having to be a bit creative with custom edit handlers to deal with snippets with foreign keys at the moment ;)

**Reply by kaedroho** [reply]
> Great, thanks, I'll look into that. Is this one image per language or just one image with translated titles?

It would copy the image, and the underlying file too.

For image alt text specifically, I'd recommend adding an alt text field on the page model that references the image. This would allow the alt text to be set based on the context of the page as well being translated with the page content.

**Reply by enzedonline** [reply]
This is kind of what I'm doing doing just now - using a struct block for all image choosing which has an image chooser field and an SEO title field. Well, it forces the editors to put in something meaningful rather than the default filename I guess ;)
